File: February2022/C++/4SUMII.cpp
class Solution {
public:
    int fourSumCount(vector<int>& nums1, vector<int>& nums2, vector<int>& nums3, vector<int>& nums4) {
        unordered_map<int,int> dict;
        int res=0;
        for(auto& i:nums1) for(auto& j:nums2) dict[i+j]++;
        for(auto& i:nums3) for(auto& j:nums4) if(dict[-(i+j)]) res+=dict[-(i+j)];
        
        return res;
    }
};

File: February2022/Python/4SumII.py
# TC : O(N^2)

class Solution:
    def fourSumCount(self, nums1: List[int], nums2: List[int], nums3: List[int], nums4: List[int]) -> int:
        map = defaultdict(int)
        count = 0
        for val1 in nums1:
            for val2 in nums2:
                map[val1+val2] += 1
        for val1 in nums3:
            for val2 in nums4:
                sum = -(val1+val2)
                if sum in map:
                    count += map[sum]
        return count
